app/controllers/concerns/application_multitenancy.rb

```diff
@@ -3,7 +3,9 @@ module ApplicationMultitenancy
   private
 
   def deduce_and_set_current_tenant
-    set_current_tenant(deduce_tenant)
+    tenant = deduce_tenant
+    ActsAsTenant.current_tenant = tenant
+    ActsAsTenant.test_tenant = tenant
```

Comment:
This doesn't look right. Why would we need `.test_tenant` in production?

Reply:
oh sorry, actually I made this when I fix the bug in Rspec, and yes this `test_tenant` is only relevant when used in our unit test, even though it won't affect production much

(the only relevance for `test_tenant` is only if we want to get the current_tenant, which is always set as `Current.current_tenant || test_tenant || || default_tenant`)

nevertheless, I'll set this one so that it'll change only if the RAILS_ENV is production

Reply:
Looked at the proposed newer code, `if ENV['RAILS_ENV'] == 'test'` I think we can just omit that, since assignment isn't costly.  We can just assign and reduce the use of magic string `'test'` in the code.

https://github.com/Coursemology/coursemology2/blob/2571b6c1d960e02afa382c0f2cbcd1a61c63a34c/config/initializers/action_cable_acts_as_tenant.rb#L8

Also isn't this now dead code? Need to clean up.

Reply:
roger that, reverting to the original one
as for `ActionCable`, it's being referred by some other important module such as `ApplicationMultitenancy`, and since this very function is being referred by the `ClassMethod` defined inside that module, it's better to keep it there

Reply:
In that case, wouldn't it be better to update the function definition of `set_current_tenant` to set both `.current_tenant` and `.test_tenant`?